### ISSUE TITLE
Reproducing the results

### ISSUE THREAD

**Issue by hondaathma** [BII-wushuang/Lie-Group-Motion-Prediction#4]
Hi,
I am unable to replicate the results in the paper. 
the final results reported is always higher than the results we are getting using multiple tf versions and using the same seeds.
For example, we ran motion_prediction --action discussion --visualize 0 and get 
0.32 & 0.62 & 1.10 & 1.31 & 1.87 & 2.04 & 2.20 & 2.44 
The results reported in the paper are:
HMR 0.29 0.55 0.83 0.94 1.35 1.49 1.61 1.72

The validation loss never reduced beyond  Best Validation Loss:  0.31701756060123426  in the first epoch.

Could you please let me know what tf.set_random_seed(112858) you set or if you are able to reproduce the results?

**Comment by hondaathma** [comment]
Just to update if we do --action all --visualize 0, the results are better but still not able to reproduce correctly.

discussion
& 0.30 & 0.58 & 0.89 & 0.98 & 1.36 & 1.52 & 1.70 & 1.84

**Comment by zaverichintan** [comment]
Hello, 
I was working on this. The main author told me that the random seed has to be changed to reproduce the results on the paper, but it is not reproducible at this moment.

**Comment by hondaathma** [comment]
I have another concern regarding the evaluation protocol.

**Comment by BII-wushuang** [comment]
We can add the following code at the beginning of main() for reproducibility:
```
random.seed(112858)
np.random.seed(112858)
tf.set_random_seed(112858)
```

This will ensure that the **same results** are returned for any given set of hyperparameters. 

Do note however that there are slight differences from the reported numbers as shown below. This could possibly due to different python/Tensorflow version.

### Results from Current random seed
| Time | 80ms | 160ms | 320ms | 400ms | 560ms | 640ms | 720ms | 1000ms |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| discussion | 0.31 | 0.59 | 0.89 | 0.97 | 1.33 | 1.47 | 1.62 | 1.80 |
| greeting | 0.52 | 0.84 | 1.21 | 1.34 | 1.58 | 1.54 | 1.56 | 1.66 |
| posing | 0.24 | 0.55 | 1.16 | 1.43 | 1.80 | 1.94 | 2.05 | 2.49 |
| walkingdog | 0.52 | 0.84 | 1.19 | 1.33 | 1.61 | 1.69 | 1.76 | 1.87 | 

### Results from Paper
| Time | 80ms | 160ms | 320ms | 400ms | 560ms | 640ms | 720ms | 1000ms |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| discussion | 0.29 | 0.55 | 0.83 | 0.94 | 1.35 | 1.49 | 1.61 | 1.72 |
| greeting | 0.52 | 0.85 | 1.25 | 1.40 | 1.65 | 1.62 | 1.67 | 1.73 |
| posing | 0.24 | 0.53 | 1.12 | 1.42 | 1.75 | 1.89 | 2.02 | 2.50 |
| walkingdog | 0.55 | 0.87 | 1.20 | 1.36 | 1.65 | 1.70 | 1.77 | 1.84 |

**Comment by BII-wushuang** [comment]
We can add the following code at the beginning of main() for reproducibility:
```
random.seed(112858)
np.random.seed(112858)
tf.set_random_seed(112858)
```

This will ensure **reproducibility**, _i.e._ **the same results are returned for any given set of hyperparameters**.

Do note that there are slight differences from the numbers reported in the paper as shown below. This could be due to different Python/Tensorflow version.

### Results from Current random seed
| Time | 80ms | 160ms | 320ms | 400ms | 560ms | 640ms | 720ms | 1000ms |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| discussion | 0.31 | 0.59 | 0.89 | 0.97 | 1.33 | 1.47 | 1.62 | 1.80 |
| greeting | 0.52 | 0.84 | 1.21 | 1.34 | 1.58 | 1.54 | 1.56 | 1.66 |
| posing | 0.24 | 0.55 | 1.16 | 1.43 | 1.80 | 1.94 | 2.05 | 2.49 |
| walkingdog | 0.52 | 0.84 | 1.19 | 1.33 | 1.61 | 1.69 | 1.76 | 1.87 | 

### Results from Paper
| Time | 80ms | 160ms | 320ms | 400ms | 560ms | 640ms | 720ms | 1000ms |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| discussion | 0.29 | 0.55 | 0.83 | 0.94 | 1.35 | 1.49 | 1.61 | 1.72 |
| greeting | 0.52 | 0.85 | 1.25 | 1.40 | 1.65 | 1.62 | 1.67 | 1.73 |
| posing | 0.24 | 0.53 | 1.12 | 1.42 | 1.75 | 1.89 | 2.02 | 2.50 |
| walkingdog | 0.55 | 0.87 | 1.20 | 1.36 | 1.65 | 1.70 | 1.77 | 1.84 |

**Comment by BII-wushuang** [comment]
We can add the following code at the beginning of main() for reproducibility:
```
random.seed(0)
np.random.seed(123456789)
tf.set_random_seed(123456789)
```

This will ensure **reproducibility**, _i.e._ **the same results are consistently returned for the given set of random seeds**.

**Comment by athus1990** [comment]
The updated code now does indeed reproduce the results

**Comment by zaverichintan** [comment]
The reply is after 2 years of the initial question. I am no longer working on this. Kindly make sure to help promptly. Anyway, I found the method producing still motion predictions. The updated code does not reproduce exact results.

**Comment by BII-wushuang** [comment]
> The updated code now does indeed reproduce the results

Thanks for the feedback!

**Comment by Droliven** [comment]
> The updated code now does indeed reproduce the results

After setting the random seed, I still cannot reproduce the reported results, how can I make it?

**Comment by BII-wushuang** [comment]
> > The updated code now does indeed reproduce the results
> 
> After setting the random seed, I still cannot reproduce the reported results, how can I make it?

To clarify, the results are slightly different from those in the paper because we didn't fix a set of random seeds back then. For example, if you fix the random seeds to
```
random.seed(0)
np.random.seed(123)
tf.set_random_seed(1234)
```
 
you should obtain the following results which are very close with the reported numbers.
 
### Results from current random seeds
| Action | 80ms | 160ms | 320ms | 400ms | 560ms | 640ms | 720ms | 1000ms |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| discussion | 0.30 | 0.59 | 0.91 | 0.99 | 1.37 | 1.50 | 1.62 | 1.73 |
| greeting | 0.53 | 0.86 | 1.26 | 1.41 | 1.63 | 1.60 | 1.62 | 1.64 |
| posing | 0.26 | 0.53 | 1.14 | 1.39 | 1.73 | 1.90 | 2.05 | 2.52 |
| walkingdog | 0.54 | 0.92 | 1.24 | 1.36 | 1.61 | 1.65 | 1.71 | 1.80 |
